registry/registry.go

```diff
@@ -254,13 +262,27 @@ func lookupSubtype(subtypeName resource.SubtypeName) (*resource.Subtype, bool) {
 }
 
 // FindValidServiceModels returns a list of valid models for a specified service.
-func FindValidServiceModels(rName resource.Name) []string {
-	validModels := make([]string, 0)
+func FindValidServiceModels(rName resource.Name) []resource.Model {
+	validModels := make([]resource.Model, 0)
 	for key := range RegisteredServices() {
 		if strings.Contains(key, rName.Subtype.String()) {
 			splitName := strings.Split(key, "/")
-			validModels = append(validModels, splitName[1])
+			model, err := resource.NewModelFromString(splitName[1])
+			utils.UncheckedError(err)
+			validModels = append(validModels, model)
 		}
 	}
 	return validModels
 }
+
+// ReconfigurableComponent is implemented when component/service of a robot is reconfigurable.
+type ReconfigurableComponent interface {
+	// Reconfigure reconfigures the resource
+	Reconfigure(ctx context.Context, cfg config.Component, deps Dependencies) error
+}
+
+// ReconfigurableService is implemented when component/service of a robot is reconfigurable.
+type ReconfigurableService interface {
+	// Reconfigure reconfigures the resource
+	Reconfigure(ctx context.Context, cfg config.Service) error
+}
```

Comment:
These were added here because they cause import cycles in the other places that also made sense (config and resource.)

Also, these are part of the overall workaround for RDK's current "reconfigure" system (which doesn't really reconfigure, but instead swaps out proxied objects.) I'll be writing up a blog/proposal to clean this all up within RDK itself in the not-too-distant future.